clients/src/main/java/org/apache/kafka/common/requests/RequestUtils.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.requests;
+
+import org.apache.kafka.clients.admin.AccessControlEntry;
+import org.apache.kafka.clients.admin.AccessControlEntryFilter;
+import org.apache.kafka.clients.admin.AclOperation;
+import org.apache.kafka.clients.admin.AclPermissionType;
+import org.apache.kafka.clients.admin.Resource;
+import org.apache.kafka.clients.admin.ResourceFilter;
+import org.apache.kafka.clients.admin.ResourceType;
+import org.apache.kafka.common.protocol.types.Struct;
+
+class RequestUtils {
+    static Resource resourceFromStructFields(Struct struct) {
+        byte resourceType = struct.getByte("resource_type");
+        String name = struct.getString("resource_name");
+        return new Resource(ResourceType.fromCode(resourceType), name);
+    }
+
+    static void resourceSetStructFields(Resource resource, Struct struct) {
+        struct.set("resource_type", resource.resourceType().code());
+        struct.set("resource_name", resource.name());
```

Comment:
It would be good to have constants instead of hardcoding the fields in many places.